e2etest/bgptests/bgp.go

```diff
@@ -600,6 +605,190 @@ var _ = ginkgo.Describe("BGP", func() {
 		ginkgo.Entry("IPV4 with Secret Ref set for BGPPeer CR", ipfamily.IPv4),
 		ginkgo.Entry("IPV6 with Secret Ref set for BGPPeer CR", ipfamily.IPv6))
 
+	ginkgo.DescribeTable("ServiceBGPStatus", func(ipFamily ipfamily.Family, poolAddresses []string, tweak testservice.Tweak) {
+		peers := metallb.PeersForContainers(FRRContainers, ipFamily)
+		sort.Slice(peers, func(i, j int) bool {
+			return peers[i].Name < peers[j].Name
+		})
+		peersNames := []string{}
```

Comment:
can we use sets and avoid the need for sorting here (and in general, avoid the dependency on ordering here)?

Reply:
we expect the controller to generate the status with the peers sorted, so this is validating this (and should be consistent)

Reply:
should we? Like, do we have a reason for that?

Reply:
yeah makes sense, plus it'll make our life easier when we try to figure out if it's the same status so we need to skip.

Reply:
getting back on this, as I changed my mind :p
It makes sense in our implementation to sort, I am not sure we should assume that on the testing side. I feel we are overtesting an internal implementation here.

Reply:
done